File: app/src/main/java/com/faendir/lightning_launcher/multitool/proxy/Screen.java
package com.faendir.lightning_launcher.multitool.proxy;

import android.content.Context;

/**
 * @author lukas
 * @since 04.07.18
 */
public interface Screen extends Proxy {
    void runAction(int action, String data);

    void runScript(String path, String name, String data);

    float getLastTouchX();

    float getLastTouchY();

    Context getContext();

    Container getContainerById(int id);

    Desktop getCurrentDesktop();
}

File: app/src/main/java/com/faendir/lightning_launcher/multitool/proxy/Screen.kt
package com.faendir.lightning_launcher.multitool.proxy

import android.content.Context

/**
 * @author lukas
 * @since 04.07.18
 */
interface Screen : Proxy {

    val lastTouchX: Float

    val lastTouchY: Float

    val context: Context

    val currentDesktop: Desktop
    fun runAction(action: Int, data: String?)

    fun runScript(path: String, name: String, data: String)

    fun getContainerById(id: Int): Container
}
